# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

****

In [8]:
%pylab inline
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from scipy import stats
import seaborn as sns


Populating the interactive namespace from numpy and matplotlib


In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [3]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [4]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [12]:
data_b = data[data.race == 'b'].call

In [19]:
n_b, (smin_b, smax_b), sm_b, sv_b, ss_b, sk_b = stats.describe(data_b)
sstr = 'B: mean = %6.4f, variance = %6.4f, skew = %6.4f, kurtosis = %6.4f'
print (sstr %(sm_b, sv_b, ss_b, sk_b))

B: mean = 0.0645, variance = 0.0603, skew = 3.5466, kurtosis = 10.5785


In [15]:
data_w = data[data.race == 'w'].call

In [17]:
n_w, (smin_w, smax_w), sm_w, sv_w, ss_w, sk_w = stats.describe(data_w)
sstr = 'W: mean = %6.4f, variance = %6.4f, skew = %6.4f, kurtosis = %6.4f'
print (sstr %(sm_w, sv_w, ss_w, sk_w))

mean = 0.0965, variance = 0.0872, skew = 2.7329, kurtosis = 5.4685


In [32]:
sm_diff = sm_w - sm_b
sd_diff = np.sqrt((sv_b/float(n_b)) + (sv_w/float(n_w)))
mean_diff_95 = sd_diff * 1.65
se_diff = sd_diff / np.sqrt(float(n_b))
ci_diff = (sm_diff - (1.96 * se_diff), sm_diff + (1.96 * se_diff))
sstr = 'DIFF: mean_diff = %6.4f, std_deviation = %6.4f, mean_diff_95 = %6.4f'
print (sstr % (sm_diff, sd_diff, mean_diff_95))
print ("SE:", se_diff, "CI:", ci_diff)

DIFF: mean_diff = 0.0320, std_deviation = 0.0078, mean_diff_95 = 0.0128
SE: 0.000157763825312 CI: (0.031723637757448693, 0.032342071952672462)


### Answer:
1. This is a problem of Difference of sample means distribution.
2. Yes CLT will apply for probability distribution.
3. ***H0:*** mean_w = mean_b, ***H1:*** mean_w > mean_b
4. CI and SE are calculated above.
5. According to ***H0***, for both mean to be equal, mean of diff should be less than 0.0128. The actual difference of mean is 0.032, ***H0*** is rejected.
6. As difference of mean is significance and does not occur by chance, the race/name is the most important factor in callback success.